.github/workflows/release.yml

```diff
@@ -135,17 +134,13 @@ jobs:
         uses: actions/download-artifact@fb598a63ae348fa914e94cd0ff38f362e927b741 # v3.0.0
         with:
           name: gradle-build-outputs
-          path: dist
+          path: build/repo
       - name: Upload assets
-        # Upload the artifacts and SLSA L3 provenance as assets to the existing release.
-        # The artifacts for each module are stored separately in a zip file. Note that
-        # the provenance will attest to the artifact files and not the aggregated zip file.
+        # Upload the artifacts and SLSA L3 provenance as assets to the existing
+        # release. Note that the provenance will attest to each artifact file and
+        # not the aggregated ZIP file.
         run: |
-          ARTIFACTS=""
-          for dir in dist/* ; do
-            zip -j `basename $dir`.zip $dir/${{ needs.release.outputs.version }}/*
-            ARTIFACTS="$ARTIFACTS `basename $dir`.zip"
-          done
-          gh release upload ${{ github.ref_name }} $ARTIFACTS          
+          find build/repo -regextype sed -regex '\(.*\.jar\|.*\.pom\)' | xargs zip artifacts.zip
```

Comment:
:x: Should also include `.module`, which is Gradle module metadata (see https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html)

Reply:
> x Should also include `.module`, which is Gradle module metadata (see https://docs.gradle.org/current/userguide/publishing_gradle_module_metadata.html)

Sure, but there is a problem. The `.module` metadata is not published when using `publishAllPublicationsToBuildRepository` task, but it's published using `publishToMavenLocal`. Is there a way to publish `.module` in the `build/repo` dir?

Reply:
Wait what? That's curious, let me check :)

Reply:
I just checked out your PR and executed `./gradlew pAPTBR` and I **do** have the `.module` files :thinking: 

```
build/repo
└── io
    └── micronaut
        ├── project-template
        │   ├── micronaut-project-template
        │   │   ├── 1.0.0-SNAPSHOT
        │   │   │   ├── maven-metadata.xml
        │   │   │   ├── maven-metadata.xml.md5
        │   │   │   ├── maven-metadata.xml.sha1
        │   │   │   ├── maven-metadata.xml.sha256
        │   │   │   ├── maven-metadata.xml.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.jar
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.jar.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.jar.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.jar.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.jar.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-javadoc.jar
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-javadoc.jar.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-javadoc.jar.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-javadoc.jar.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-javadoc.jar.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.module
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.module.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.module.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.module.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.module.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.pom
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.pom.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.pom.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.pom.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1.pom.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-sources.jar
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-sources.jar.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-sources.jar.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-sources.jar.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220908.110730-1-sources.jar.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.jar
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.jar.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.jar.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.jar.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.jar.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-javadoc.jar
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-javadoc.jar.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-javadoc.jar.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-javadoc.jar.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-javadoc.jar.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.module
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.module.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.module.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.module.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.module.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.pom
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.pom.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.pom.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.pom.sha256
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2.pom.sha512
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-sources.jar
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-sources.jar.md5
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-sources.jar.sha1
        │   │   │   ├── micronaut-project-template-1.0.0-20220922.075540-2-sources.jar.sha256
        │   │   │   └── micronaut-project-template-1.0.0-20220922.075540-2-sources.jar.sha512
        │   │   ├── maven-metadata.xml
        │   │   ├── maven-metadata.xml.md5
        │   │   ├── maven-metadata.xml.sha1
        │   │   ├── maven-metadata.xml.sha256
        │   │   └── maven-metadata.xml.sha512
        │   └── micronaut-project-template-bom
        │       ├── 1.0.0-SNAPSHOT
        │       │   ├── maven-metadata.xml
        │       │   ├── maven-metadata.xml.md5
        │       │   ├── maven-metadata.xml.sha1
        │       │   ├── maven-metadata.xml.sha256
        │       │   ├── maven-metadata.xml.sha512
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.module
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.module.md5
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.module.sha1
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.module.sha256
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.module.sha512
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.pom
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.pom.md5
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.pom.sha1
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.pom.sha256
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.pom.sha512
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.toml
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.toml.md5
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.toml.sha1
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.toml.sha256
        │       │   ├── micronaut-project-template-bom-1.0.0-20220908.110730-1.toml.sha512
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.module
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.module.md5
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.module.sha1
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.module.sha256
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.module.sha512
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.pom
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.pom.md5
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.pom.sha1
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.pom.sha256
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.pom.sha512
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.toml
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.toml.md5
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.toml.sha1
        │       │   ├── micronaut-project-template-bom-1.0.0-20220922.075540-2.toml.sha256
        │       │   └── micronaut-project-template-bom-1.0.0-20220922.075540-2.toml.sha512
        │       ├── maven-metadata.xml
        │       ├── maven-metadata.xml.md5
        │       ├── maven-metadata.xml.sha1
        │       ├── maven-metadata.xml.sha256
        │       └── maven-metadata.xml.sha512
        └── xxx
            ├── micronaut-xxx
            │   ├── 1.0.0-SNAPSHOT
            │   │   ├── maven-metadata.xml
            │   │   ├── maven-metadata.xml.md5
            │   │   ├── maven-metadata.xml.sha1
            │   │   ├── maven-metadata.xml.sha256
            │   │   ├── maven-metadata.xml.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.jar
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.jar.md5
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.jar.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.jar.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.jar.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-javadoc.jar
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-javadoc.jar.md5
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-javadoc.jar.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-javadoc.jar.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-javadoc.jar.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.module
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.module.md5
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.module.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.module.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.module.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.pom
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.pom.md5
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.pom.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.pom.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1.pom.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-sources.jar
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-sources.jar.md5
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-sources.jar.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-sources.jar.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211006.123715-1-sources.jar.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.jar
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.jar.md5
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.jar.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.jar.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.jar.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-javadoc.jar
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-javadoc.jar.md5
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-javadoc.jar.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-javadoc.jar.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-javadoc.jar.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.module
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.module.md5
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.module.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.module.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.module.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.pom
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.pom.md5
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.pom.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.pom.sha256
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2.pom.sha512
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-sources.jar
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-sources.jar.md5
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-sources.jar.sha1
            │   │   ├── micronaut-xxx-1.0.0-20211202.123441-2-sources.jar.sha256
            │   │   └── micronaut-xxx-1.0.0-20211202.123441-2-sources.jar.sha512
            │   ├── maven-metadata.xml
            │   ├── maven-metadata.xml.md5
            │   ├── maven-metadata.xml.sha1
            │   ├── maven-metadata.xml.sha256
            │   └── maven-metadata.xml.sha512
            └── micronaut-xxx-bom
                ├── 1.0.0-SNAPSHOT
                │   ├── maven-metadata.xml
                │   ├── maven-metadata.xml.md5
                │   ├── maven-metadata.xml.sha1
                │   ├── maven-metadata.xml.sha256
                │   ├── maven-metadata.xml.sha512
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.module
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.module.md5
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.module.sha1
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.module.sha256
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.module.sha512
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.pom
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.pom.md5
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.pom.sha1
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.pom.sha256
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.pom.sha512
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.toml
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.toml.md5
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.toml.sha1
                │   ├── micronaut-xxx-bom-1.0.0-20211202.123441-1.toml.sha256
                │   └── micronaut-xxx-bom-1.0.0-20211202.123441-1.toml.sha512
                ├── maven-metadata.xml
                ├── maven-metadata.xml.md5
                ├── maven-metadata.xml.sha1
                ├── maven-metadata.xml.sha256
                └── maven-metadata.xml.sha512

12 directories, 185 files
```

(btw you can add `.toml` in the artifact types, those are Gradle version catalogs)